### ISSUE TITLE
UI problems with "connected to" counters

### ISSUE THREAD

**Issue by josellm** [metabase/metabase#6830]
Hi,

I have a customers table related to eleven tables. At detail view of a row sometimes the related data is not loading.  If the related data is slow to load and I click at navigate bottons(right and left arrows) I can view the older values of the last data and If I receive the response of the older related data the number is writed and replaced with the actually related data.

- Your browser and the version: last Chrome and last Firefox
- Your operating system: windows 7
- Your databases: SQL Server
- Metabase version: 0.27.2
- Metabase hosting environment: Java jar file
- Metabase internal database:  H2

- Repeatable steps to reproduce the issue:

Go to detail of a row with slowly connected data, navigate to next row.

I think that a correct behaivor is:

- Clear connected data values on load detail page
- Cancel pending querys when navigate to next page
- Link to detail connected data before number is loading
- Paint data values independently and not when all responses are finished

Metabase is a awesome project! 

Thanks!

**Comment by satyadeepk** [comment]
We are also facing this. 

Metabase version: 0.28.4
Setup: Docker on EC2
Internal DB: Postgres on RDS
Query DB: MySQL

On first load of detail view, this exception occurs. (All API calls seems to succeed though in network tab of chrome inspector).

```
Uncaught (in promise) TypeError: Cannot read property 'length' of null
    at _callee21$ (actions.js:1349)
    at tryCatch (runtime.js:65)
    at Generator.invoke [as _invoke] (runtime.js:303)
    at Generator.prototype.(anonymous function) [as next] (https://metabase.sheroes.com/app/dist/vendor.bundle.js?b50577193a4885b9a906:118381:21)
    at step (actions.js:75)
    at actions.js:75
    at new Promise (<anonymous>)
    at actions.js:75
    at actions.js:1346
    at _callee$ (redux.js:19)
```
![image](https://user-images.githubusercontent.com/478578/38166599-f8416ff8-3543-11e8-9d80-a1797ed3bb93.png)

But after pressing the Refresh button in the middle of the screen, the counter values load fine.

![image](https://user-images.githubusercontent.com/478578/38166610-2df6f938-3544-11e8-93e7-9949a1d7838d.png)

**Comment by modille** [comment]
Same here on `Metabase version v0.29.0 (ff2120a release-0.29.0)`

After first load of the detail view I see the spinners, and the logs show:

```
05-17 11:00:46 DEBUG metabase.middleware :: POST /api/dataset 200 (31 ms) (25 DB calls). Jetty threads: 8/50 (4 busy, 6 idle, 0 queued)
05-17 11:00:46 DEBUG metabase.middleware :: GET /api/table/75/fks 200 (3 ms) (4 DB calls). Jetty threads: 8/50 (4 busy, 6 idle, 0 queued)
```

Then after clicking refresh the counter values load fine and the logs show:

```
05-17 11:00:58 DEBUG metabase.middleware :: POST /api/dataset 200 (25 ms) (25 DB calls). Jetty threads: 8/50 (4 busy, 6 idle, 0 queued)
05-17 11:00:58 WARN middleware.expand :: Referring to fields by their bare ID (134) is deprecated in MBQL '98. Please use [:field-id 134] instead.
05-17 11:00:58 WARN middleware.expand :: Referring to fields by their bare ID (134) is deprecated in MBQL '98. Please use [:field-id 134] instead.
05-17 11:00:58 WARN middleware.expand :: You shouldn't specify an :and filter with only one subclause.
05-17 11:00:58 DEBUG metabase.middleware :: POST /api/dataset 200 (33 ms) (15 DB calls). Jetty threads: 8/50 (4 busy, 6 idle, 0 queued)
05-17 11:00:58 WARN middleware.expand :: Referring to fields by their bare ID (151) is deprecated in MBQL '98. Please use [:field-id 151] instead.
05-17 11:00:58 WARN middleware.expand :: Referring to fields by their bare ID (151) is deprecated in MBQL '98. Please use [:field-id 151] instead.
05-17 11:00:58 WARN middleware.expand :: You shouldn't specify an :and filter with only one subclause.
05-17 11:00:58 DEBUG metabase.middleware :: POST /api/dataset 200 (25 ms) (15 DB calls). Jetty threads: 8/50 (4 busy, 6 idle, 0 queued)
05-17 11:00:58 WARN middleware.expand :: Referring to fields by their bare ID (157) is deprecated in MBQL '98. Please use [:field-id 157] instead.
05-17 11:00:58 WARN middleware.expand :: Referring to fields by their bare ID (157) is deprecated in MBQL '98. Please use [:field-id 157] instead.
05-17 11:00:58 WARN middleware.expand :: You shouldn't specify an :and filter with only one subclause.
05-17 11:00:58 DEBUG metabase.middleware :: POST /api/dataset 200 (26 ms) (15 DB calls). Jetty threads: 8/50 (4 busy, 6 idle, 0 queued)
05-17 11:00:58 WARN middleware.expand :: Referring to fields by their bare ID (193) is deprecated in MBQL '98. Please use [:field-id 193] instead.
05-17 11:00:58 WARN middleware.expand :: Referring to fields by their bare ID (193) is deprecated in MBQL '98. Please use [:field-id 193] instead.
05-17 11:00:58 WARN middleware.expand :: You shouldn't specify an :and filter with only one subclause.
05-17 11:00:58 DEBUG metabase.middleware :: POST /api/dataset 200 (16 ms) (15 DB calls). Jetty threads: 8/50 (4 busy, 6 idle, 0 queued)
```

**Comment by paulrosenzweig** [comment]
Resolved by #10792
